src/PSRule.Rules.Azure/rules/Azure.Deployment.Rule.ps1

```diff
@@ -55,4 +60,52 @@ function global:RecurseDeploymentSensitive {
     }
 }
 
+# Check resource properties that should be set by secure parameters.
+function global:RecurseSecureValue {
+    param (
+        [Parameter(Mandatory = $True)]
+        [PSObject]$Deployment
+    )
+    process {
+        $resources = @($Deployment.properties.template.resources);
+        if ($resources.Length -eq 0) {
+            return $Assert.Pass();
+        }
+
+        $secureParameters = @($Deployment.properties.template.parameters.PSObject.properties | Where-Object {
+            $_.Value.type -eq 'secureString' -or $_.Value.type -eq 'secureObject'
+        } | ForEach-Object {
+            $_.Name
+        });
+        Write-Debug -Message "Secure parameters are: $($secureParameters -join ', ')";
+
+        foreach ($resource in $resources) {
+            if ($resource.type -eq 'Microsoft.Resources/deployments') {
+                RecurseSecureValue -Deployment $resource;
+            }
+            elseif ($resource.type -eq 'Microsoft.Compute/virtualMachineScaleSets') {
+                if ($resource.properties.virtualMachineProfile.osProfile.adminPassword) {
+                    $isSecure = [PSRule.Rules.Azure.Runtime.Helper]::HasValueFromSecureParameter($resource.properties.virtualMachineProfile.osProfile.adminPassword, $secureParameters);
```

Comment:
Ohh, this was my main question, how to know from the rule code that the property was set with a parameter, and get information about it, because while debugging I only saw the final replaced value of the parameters under, for example, $TargetObject.properties.virtualMachineProfile.osProfile.adminPassword, so I didn't know how I could even try to extract the parameter name. I suppose I made a mistake debugging and then didn't express myself properly when asking D: There was no need to do all this, I was planning on doing it myself once I understood that. Thank you very much anyways, it looks very good! Is it okay if I add the rest of the properties to check to this same PR?